packages/suite/src/actions/suite/metadataActions.ts

```diff
@@ -348,8 +351,16 @@ export const fetchMetadata =
             throw new Error('no provider instance');
         }
 
-        const { fileName, aesKey } = entity[encryptionVersion];
-        console.debug('metadata: fatching file: ', fileName);
+        if (entity.type === 'device' && entity.status !== 'enabled') {
+            return;
+        }
+
+        const entityData = entity[encryptionVersion];
+        if (!entityData) {
+            return;
+        }
```

Comment:
this makes me think "what does `entityData` mean?"
what about just?
```
if (!entity[encryptionVersion]) return
```

Reply:
but then there is `const { fileName, aesKey } = entityData;` on the next line anyways